tests/unit/s2n_x509_validator_certificate_signatures_test.c

```diff
@@ -49,115 +54,12 @@ int main(int argc, char **argv)
         .signature_schemes = test_sig_scheme_list,
     };
 
-    const struct s2n_signature_scheme *const pss_sig_scheme_list[] = {
-        &s2n_rsa_pss_pss_sha256,
-        &s2n_rsa_pss_pss_sha384,
-        &s2n_rsa_pss_pss_sha512,
-        &s2n_rsa_pss_rsae_sha256,
-        &s2n_rsa_pss_rsae_sha384,
-        &s2n_rsa_pss_rsae_sha512,
-    };
-
-    const struct s2n_signature_preferences pss_certificate_signature_preferences = {
-        .count = s2n_array_len(pss_sig_scheme_list),
-        .signature_schemes = pss_sig_scheme_list,
+    const struct s2n_security_policy test_sp = {
+        .minimum_protocol_version = S2N_TLS12,
+        .certificate_signature_preferences = &test_certificate_signature_preferences,
     };
 
-    /* s2n_is_certificate_sig_scheme_supported() */
```

Comment:
Shouldn't all of these tests work with `s2n_validator_check_cert_preferences()`? I see you've added new tests for `s2n_security_policy_validate_sig_scheme_supported()`, but having tests for `s2n_validator_check_cert_preferences()` with real certs could be useful, unless you don't think they're valuable anymore.

Reply:
It can be dangerous to remove tests during a refactor. You might miss a behavior change.
If the setup is repetitive, can you fix that without removing the test?